Mage/src/main/java/mage/abilities/effects/ContinuousEffects.java

```diff
@@ -523,97 +522,142 @@ public ApprovingObject asThough(UUID objectId, AsThoughEffectType type, Ability
-                        }
                     }
+
+                    approvingObjects.add(new ApprovingObject(ability, game));
+                    abilityAsThoughEffectMap.put(ability, effect);
                 }
             }
 
-            if (possibleApprovingObjects.size() == 1) {
-                return possibleApprovingObjects.iterator().next();
-            } else if (possibleApprovingObjects.size() > 1) {
-                // Select the ability that you use to permit the action                
-                Map<String, String> keyChoices = new HashMap<>();
-                for (ApprovingObject approvingObject : possibleApprovingObjects) {
-                    MageObject mageObject = game.getObject(approvingObject.getApprovingAbility().getSourceId());
-                    String choiceKey = approvingObject.getApprovingAbility().getId().toString();
-                    String choiceValue;
-                    if (mageObject == null) {
-                        choiceValue = approvingObject.getApprovingAbility().getRule();
-                    } else {
-                        choiceValue = mageObject.getIdName() + ": " + approvingObject.getApprovingAbility().getRule(mageObject.getName());
-                    }
-                    keyChoices.put(choiceKey, choiceValue);
+            Set<ApprovingObject> tmpApprovingObjects = (Set<ApprovingObject>) game.getState().getValue(keyAprovingObjects);
+            if (tmpApprovingObjects != null) {
+                approvingObjects.addAll(tmpApprovingObjects);
+            }
+            game.getState().setValue(keyAprovingObjects, approvingObjects);
+
+            Map<Ability, AsThoughEffect> tmpAbilityAsThoughEffectMap = (Map<Ability, AsThoughEffect>) game.getState().getValue(keyAbilityAsThoughEffectMap);
+            if (tmpAbilityAsThoughEffectMap != null) {
+                abilityAsThoughEffectMap.putAll(tmpAbilityAsThoughEffectMap);
+            }
+            game.getState().setValue(keyAbilityAsThoughEffectMap, abilityAsThoughEffectMap);
+
+            // If only checking for playable, return the first available choice.
+            // The choice doesn't matter as long as something makes it playable.
+            if (!approvingObjects.isEmpty()) {
+                return approvingObjects.iterator().next();
+            } else {
+                return null;
+            }
+        } else {
+            // Load values back from state saved during the pass-through when game.inCheckPlayableState() was true
+            approvingObjects = (Set<ApprovingObject>) game.getState().getValue(keyAprovingObjects);
+            game.getState().removeValue(keyAprovingObjects); // TODO: Where is the right place to remove them?
+
+            abilityAsThoughEffectMap = (Map<Ability, AsThoughEffect>) game.getState().getValue(keyAbilityAsThoughEffectMap);
+            game.getState().removeValue(keyAbilityAsThoughEffectMap); // TODO: Where is the right place to remove them?
```

Comment:
I need to remove the saved values from the state at some point but I am not sure when.
Here is certainly the wrong time, since they are still needed for mana payment.